lib/perfdata/influxdbcommonwriter.hpp

```diff
@@ -50,14 +48,17 @@ class InfluxdbCommonWriter : public ObjectImpl<InfluxdbCommonWriter>
 
 private:
 	Timer::Ptr m_FlushTimer;
+	WorkQueue m_WorkQueue{10000000, 1};
+	std::vector<String> m_DataBuffer;
+	std::atomic_size_t m_DataBufferSize;
```

Comment:
Huh? No deterministic init value? (Hint: `icinga::Atomic<T>`)

Reply:
Indeed, that one isn't initialized properly. But I don't get why `icinga::Atomic` would improve anything compared to `std::atomic`.

Reply:
It has an atomic constructor in addition.

Reply:
What's an atomic constructor even supposed to do? If the code decides to construct the same object concurrently, I think that's the issue, not that a constructor wouldn't handle it (I'm not even sure if it would actually be possible to handle this correctly). If I understand https://en.cppreference.com/w/cpp/atomic/atomic/atomic correctly, if you use that default constructor of `std::atomic`, you'd actually have to call `std::atomic_init()` on it to initialize it properly which `icinga::Atomic` doesn't do.

Reply:
Rationale: call store() at construction time, i.e. before anyone else calls any other method, so that the instance is well initialised before usage. Doesn’t

```cpp
std::atomic<T> x;
x.store(y);
```

initialises x properly?

Reply:
See the "Notes" section on https://en.cppreference.com/w/cpp/atomic/atomic/atomic:

> The default-initialized `std::atomic<T>` does not contain a `T` object, and its only valid uses are destruction and initialization by `std::atomic_init`, see LWG 2334

So I think the call to `x.store(y);` is incorrect when strictly following the standard without calling `std::atomic_init(&x, 0);` first.

Reply:
But atomic_init "is not atomic" (https://en.cppreference.com/w/cpp/atomic/atomic_init). At least store() is.

Reply:
Why would you need this to be atomic in the first place? What should access the object concurrently while constructing it?

Reply:
I believe an atomic operation shall only follow an atomic operation, so Atomic<T> performs atomic construction.

Reply:
Also: why store() shouldn’t _just do it_?